pkg/manager/member/pd_scaler.go

```diff
@@ -121,13 +121,14 @@ func (s *pdScaler) ScaleIn(meta metav1.Object, oldSet *apps.StatefulSet, newSet
 	// If the pd pod was pd leader during scale-in, we would transfer pd leader to pd-0 directly
 	// If the pd statefulSet would be scale-in to zero and the pd-0 was going to be deleted,
 	// we would directly deleted the pd-0 without pd leader transferring
-	if ordinal > 0 {
+	minOrdinal := helper.GetMinPodOrdinal(*newSet.Spec.Replicas, newSet)
+	if ordinal > minOrdinal {
```

Comment:
Sorry that I missed one point that with advanced statefulset we can scale in any Pod, so the logic should be :
```
if *newSet.Spec.Replicas > 1
    if ordinal > minOrdinal, transfer to minOrdinal
    else, transfer to maxOrdinal
else, no need to transfer leader
```

Reply:
I think we need check `leader.Name == memberName` first:
```
if leader.Name == memberName 
    if ordinal > minOrdinal, transfer to minOrdinal
    else, transfer to maxOrdinal
```

What do you think?

Reply:
> Sorry that I missed one point that with advanced statefulset we can scale in any Pod, so the logic should be :
> 
> ```
> if *newSet.Spec.Replicas > 1
>     if ordinal > minOrdinal, transfer to minOrdinal
>     else, transfer to maxOrdinal
> else, no need to transfer leader
> ```

I think we need check `leader.Name == memberName `first:
```
if leader.Name == memberName 
    if ordinal > minOrdinal, transfer to minOrdinal
    else, transfer to maxOrdinal
```
What do you think?

Reply:
What about:
```
if *newSet.Spec.Replicas > 1
    if leader.Name == memberName 
        if ordinal > minOrdinal, transfer to minOrdinal
        else, transfer to maxOrdinal
else, no need to transfer leader
```

Reply:
That's what I meant.